lib/gcloud/datastore/dataset/query_results.rb

```diff
@@ -133,7 +133,7 @@ def initialize arr = []
         #   end
         #
         def next?
-          !no_more?
+          not_finished?
```

Comment:
If this is being exposed to the user, it should probably be the same.

Our clients only should care about MORE_RESULTS_AFTER_LIMIT vs NO_MORE_RESULTS.

They should never see NOT_FINISHED, because we will always keep querying until it is not NOT_FINISHED